File: blog/20150720/jpa_oneToMany_b/src/main/java/sample/model/ProductVariation.java
package sample.model;

import lombok.Data;
import lombok.ToString;

import javax.persistence.*;

@Data
@ToString(exclude = "product")
@Entity
@Table(name = "product_variation")
public class ProductVariation {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String color;
	private String size;

	@ManyToOne
	private Product product;
}

File: blog/20150727/querydsl-scala-jpa-sample/src/main/scala-model/sample/model/ProductVariation.scala
package sample.model

import javax.persistence._

@Entity
@Table(name = "product_variation")
class ProductVariation {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	var id: Long = _
	var color: String = _
	var size: String = _

	override def toString = s"ProductVariation(id: ${id}, color: ${color}, size: ${size})"
}
